tests/functests/instancetype_test.go

```diff
@@ -437,3 +472,54 @@ func expectSSHToRunCommand(virtClient kubecli.KubevirtClient, vmName, username s
 		g.Expect(err).ToNot(HaveOccurred())
 	}, defaultVMReadyTimeout, 10*time.Second).Should(Succeed())
 }
+
+func GetDeprecatedFieldFromCrd(virtClient kubecli.KubevirtClient, crdVersion string) []string {
+	crd, err := virtClient.ExtensionsClient().ApiextensionsV1().CustomResourceDefinitions().Get(
+		context.Background(), crdVersion, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	var nonDeprecatedVersion apiextensionsv1.CustomResourceDefinitionVersion
+	for _, version := range crd.Spec.Versions {
+		if !version.Deprecated {
+			nonDeprecatedVersion = version
+		}
+	}
+	schema := nonDeprecatedVersion.Schema.OpenAPIV3Schema
+
+	var deprecatedFields []string
+	findDeprecatedFields(schema, "", &deprecatedFields)
+	return deprecatedFields
+}
+
+func findDeprecatedFields(schema *apiextensionsv1.JSONSchemaProps, path string, result *[]string) {
+	for propName, propSchema := range schema.Properties {
+		fullPath := propName
+		if path != "" {
+			fullPath = path + "." + propName
+		}
+		if strings.Contains(strings.ToLower(propSchema.Description), "deprecated") {
+			*result = append(*result, fullPath)
+		}
+
+		findDeprecatedFields(&propSchema, fullPath, result)
+	}
+}
+
+func isFieldUsed(obj interface{}, path string) bool {
```

Comment:
What about constructing jsonpaths in `findDeprecatedFields` and trying to extract them here instead of using reflection? 

Like used here:

https://github.com/kubevirt/kubevirt/blob/2c03eec040e7844046726c949af3bd868c4144f9/pkg/virt-operator/resource/generate/components/crds_test.go#L352

Reply:
Wouldn't it be more appropriate to use Go's native reflect package?
I’d assume using a third-party library is less preferable than relying on the standard library.

Reply:
Perhaps but reflect can also be tricky to get right.